src/view/Animations.hs

```diff
@@ -0,0 +1,55 @@
+module View.Animations where
+
+import Graphics.Gloss
+import Model.Movement
+import Model.GameState
+import View.View
+
+-- This module contains animations which play when certain events happen.
+
+-- Take the animations from the queue and play the right type of animation at the right position.
+handleAnimationQueue :: GameState -> IO GameState
+handleAnimationQueue gs@GameState{animations = []} = return gs -- Do nothing is there are no animations to be played.
+handleAnimationQueue gs@GameState{animations = (a1:as)} =
+    do case animationType a1 of -- Otherwise, playe the next animation and the rest recursively.
+        PowerUpAnimation -> animate window black (playAnimation PowerUpAnimation(animationPos a1))
+        BulletAnimation -> animate window black (playAnimation BulletAnimation  (animationPos a1))
+        _ -> animate window black (playAnimation DespawnAnimation (animationPos a1))
```

Comment:
Volgens mij heb je deze hele case niet nodig, want je doet in principe hetzelfde voor elk animatietype: je geeft het mee aan playAnimation.